File: frontend/src/app/core/services/review.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';

export interface Review {
  id: string;
  reviewer: { id: string; full_name: string; username: string; avatar_url: string };
  rating: number;
  comment: string;
  delivery_rating: number | null;
  communication_rating: number | null;
  accuracy_rating: number | null;
  seller_response: string | null;
  seller_responded_at: string | null;
  created_at: string;
}

export interface ReviewStats {
  average: number;
  total: number;
  distribution: Record<number, number>;
  avg_delivery: number;
  avg_communication: number;
  avg_accuracy: number;
}

@Injectable({ providedIn: 'root' })
export class ReviewService {
  constructor(private api: ApiService) {}

  getSellerReviews(userId: string): Observable<{ reviews: any; stats: ReviewStats }> {
    return this.api.get(`users/${userId}/reviews`);
  }

  createReview(data: { seller_id: string; transaction_id?: string; rating: number; comment?: string; delivery_rating?: number; communication_rating?: number; accuracy_rating?: number }): Observable<any> {
    return this.api.post('reviews', data);
  }

  respondToReview(reviewId: string, response: string): Observable<any> {
    return this.api.post(`reviews/${reviewId}/respond`, { response });
  }
}

File: flutter_app/lib/services/review_service.dart
import 'api_service.dart';

class ReviewService {
  final ApiService _api;

  ReviewService(this._api);

  Future<Map<String, dynamic>> createReview({
    required String sellerId,
    required int rating,
    String? comment,
    String? transactionId,
  }) async {
    final response = await _api.post('/reviews', data: {
      'seller_id': sellerId,
      'rating': rating,
      if (comment != null) 'comment': comment,
      if (transactionId != null) 'transaction_id': transactionId,
    });
    return response.data;
  }

  Future<Map<String, dynamic>> getSellerReviews(String sellerId) async {
    final response = await _api.get('/reviews/seller/$sellerId');
    return response.data is Map<String, dynamic> ? response.data as Map<String, dynamic> : {};
  }

  Future<void> respondToReview(String reviewId, String response) async {
    await _api.post('/reviews/$reviewId/respond', data: {
      'response': response,
    });
  }
}
